### PR TITLE
feat: add flag for close

### PR DESCRIPTION
**Pull request by jemmyshin**
We add a flag for denoting whether `Annlite` has been closed already to avoid closing it twice.

### CODE REVIEW

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: annlite/storage/kv.py [hunk 1]
```diff
@@ -98,8 +100,16 @@ def clear(self):
         self._init_db(create_if_missing=True, **self._kwargs)
 
     def close(self):
+        if self._is_closed:
```
Comment (numb3r3) [inline annlite/storage/kv.py:103]: revert the changes here

**Review by numb3r3 — APPROVED** [review]

**Comment by JoanFM** [comment]
not a big fan of adding such a flag? why is this needed?

**Review by numb3r3 — CHANGES_REQUESTED** [review]
@jemmyshin add unittest

**Comment by numb3r3** [comment]
@JoanFM In the current docarray, one instance of annlite would be closed (i.e., close rocksdb session) two times some times. Rocksdb does not allow closing a closed session. Hence, in annlite, we use a flag to overcome this problem.

File: annlite/storage/kv.py [hunk 2]
```diff
@@ -91,15 +94,26 @@ def get(self, doc_ids: Union[str, list]) -> DocumentArray:
         return docs
 
     def clear(self):
-        self._db.close()
+        if self._is_closed:
+            warnings.warn(
```
Comment (numb3r3) [inline annlite/storage/kv.py:98]: no warning, just pass

**Review by numb3r3 — CHANGES_REQUESTED** [review]

**Comment by JoanFM** [comment]
> @JoanFM In the current docarray, one instance of annlite would be closed (i.e., close rocksdb session) two times some times. Rocksdb does not allow closing a closed session. Hence, in annlite, we use a flag to overcome this problem.

Isn't there an API from rocksdb to check if session has been closed?

**Comment by numb3r3** [comment]
no such API from rocksdb

**Pull request merged**